crates/host/src/wasmbus/mod.rs

```diff
@@ -492,31 +281,44 @@ impl Handler {
                 self.lattice_prefix, invocation.target.public_key
             ),
         };
+
+        let timeout = chunked.then_some(CHUNK_RPC_EXTRA_TIME); // TODO: add rpc_nats timeout
+        let request = Request::new().payload(payload.into()).timeout(timeout);
         let res = self
             .nats
-            .request(topic, request.into())
+            .send_request(topic, request)
             .await
             .context("failed to publish on NATS topic")?;
+
         let InvocationResponse {
             invocation_id,
-            msg,
+            mut msg,
             content_length,
             error,
+            ..
         } = rmp_serde::from_slice(&res.payload).context("failed to decode invocation response")?;
         ensure!(invocation_id == invocation.id, "invocation ID mismatch");
-        if let Some(content_length) = content_length {
-            let content_length =
-                usize::try_from(content_length).context("content length does not fit in usize")?;
-            ensure!(content_length == msg.len(), "message size mismatch");
+
+        let resp_length = usize::try_from(content_length.unwrap_or_default())
+            .context("content length does not fit in usize")?;
+        if needs_chunking(resp_length) {
+            msg = self
+                .chunk_endpoint
+                .get_unchunkified(&invocation_id)
+                .await
+                .context("failed to dechunk response")?;
```

Comment:
There's no need to call `needs_chunking` with a `content_length` set to `None`, so please bring the `if let Some....` block back

Reply:
Doesn't it depend on provider implementation?
If we expect `content_length` to always be set here, then I'm fine with the type change and failing when content length is not set

Reply:
actually, since we enforce that the payload length is `0` if the field is not set, I'm ok with this

Reply:
No, invocations/invocation responses shouldn't be manually constructed by a provider implementation; it should be common to the provider SDK, e.g. https://github.com/wasmCloud/wasmCloud/blob/main/crates/provider-sdk/src/rpc_client.rs#L178

I changed the type, and tested with legacy HTTP and Redis KV providers 👍